### PR TITLE
Update upkeep state store to match updated interface

### PR DESCRIPTION
**Pull request by EasterTheBunny**
The upkeep state store reader and writer interfaces were changed to provide workIDs and check results respectively. The implementation was updated to match the requirements of the new interfaces.

As of right now, `SetUpkeepState` ignores the provided state in favor of assuming state from the result eligibility.

### CODE REVIEW

File: core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store.go [hunk 1]
```diff
@@ -1,97 +1,95 @@
...
+)
 
 type UpkeepStateStore struct {
-	mu         sync.RWMutex
-	statesByID map[string]*upkeepState
-	states     []*upkeepState
-	lggr       logger.Logger
+	mu        sync.RWMutex
+	workIDIdx map[string]*upkeepState
+	states    []*upkeepState
```
Comment (amirylm) [inline core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store.go:31]: a single map should be sufficient

File: core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store_test.go [hunk 2]
```diff
@@ -1,134 +1,137 @@
...
+		assert.NoError(t, store.SetUpkeepState(context.Background(), insert, ocr2keepers.Performed), "storing states should not produce an error")
+	}
+
+	states, err := store.SelectByWorkIDs(workIDs...)
+	assert.NoError(t, err, "no error expected from selecting states")
+
+	assert.Equal(t, expected, states, "upkeep state values should match expected")
+}
+
+func TestUpkeepStateStore_SetUpkeepState(t *testing.T) {
```
Comment (amirylm) [inline core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store_test.go:62]: Use test cases with input/output and expected assertions
```
tests := []struct{ 
name string
input ocr2keepers.CheckResult
...
```

File: core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store.go (hunk 1 above)
Comment (EasterTheBunny) [reply core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store.go:31]: Keep in mind this is very temporary. I'll be doing a fast follower with the actual database implementation.

Comment (EasterTheBunny) [reply core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store.go]: done

**Review by amirylm — DISMISSED** [review]

**Review by amirylm — APPROVED** [review]

**Pull request merged**